src/shared/bpf-program.c

```diff
@@ -325,7 +326,18 @@ int bpf_map_new(
         for (size_t i = 0; i < sizeof(attr.map_name) - 1 && *n; i++, n++)
                 attr.map_name[i] = strchr(ALPHANUMERICAL ".", *n) ? *n : '_';
 
-        return RET_NERRNO(bpf(BPF_MAP_CREATE, &attr, sizeof(attr)));
+        fd = RET_NERRNO(bpf(BPF_MAP_CREATE, &attr, sizeof(attr)));
+        if (fd < 0)
+                return fd;
+
+        /* Work around libbpf bug where it doesn't unsets O_CLOEXEC:
+         * https://github.com/libbpf/libbpf/issues/812 */
+        r = fd_cloexec(fd, true);
+        if (r < 0)
+                return r;
+        if (r > 0)
+                log_warning("bpf(BPF_MAP_CREATE) returned an fd without O_CLOEXEC, fixing.");
```

Comment:
maybe 'info' is more appropriate, given there's nothing the user can do, and it is worked around?